Pod/Classes/URLSession+Extensions.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  NSURLSession+Extensions.swift
+//  Pods
+//
+//  Created by Matt Thomas on 3/17/16.
+//
+//
+import Foundation
+
+public typealias JSON = [AnyHashable: Any]
+
+public enum APIClientError: Error {
+    case cannotCreateURL
+    case badResponse(response: URLResponse)
+    case noData(response: URLResponse?)
+    case cannotSerializeToJSON(data: Data)
+    case resultJSONMissingKey(key: String, json: JSON)
```

Comment:
only `badResponse` and `no data` are used by the `URLSession` extension... do you want to leave the other cases out of this enum?